mod/cp_notifications/actions/cp_notifications/group_mail.php

```diff
@@ -0,0 +1,105 @@
+<?php
+
+/**
+* Group Tools
+* Mail all the members of a group
+* 
+* @author ColdTrick IT Solutions
+* Modified by Christine Yu
+**/	
+
+$group_guid = (int) get_input("group_guid", 0);
+$user_guids = get_input("user_guids");
+$subject = get_input("title"); 
+$body = get_input("description");
+
+$forward_url = REFERER;
+
+$user_guids = group_tools_retrieve_members($group_guid, $user_guids);
+
+if (!empty($group_guid) && !empty($body) && !empty($user_guids)) {
+	$group = get_entity($group_guid);
+	
+	if (!empty($group) && ($group instanceof ElggGroup)) {
+		if ($group->canEdit()) {
+			set_time_limit(0);
+			
+			$body .= PHP_EOL . PHP_EOL;
+			$body .= elgg_echo("group_tools:mail:message:from") . ": {$group->name} [{$group->getURL()}]";
+			
+			// cyu - 03/07/2016: we need to use the cp_notifications module to handle the improved notifications
+			if (elgg_is_active_plugin('cp_notifications')) {
```

Comment:
Since this code is inside the `cp_notifications` module, can this actually ever be false?

Reply:
I basically just made an exact carbon copy of the original file, then slightly modified it